dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java

```diff
@@ -198,4 +213,42 @@ private static File findLocalRepository() {
     return managedDependencies;
   }
 
+  /**
+   * Configures Maven repositories to search for dependencies.
+   *
+   * @param addMavenCentral if true, add Maven Central to the end of the repository list
+   */
+  public static void configureMavenRepositories(Iterable<String> mavenRepositoryUrls,
+      boolean addMavenCentral) {
+    ImmutableList.Builder<RemoteRepository> repositoryListBuilder = ImmutableList.builder();
+    for (String mavenRepositoryUrl : mavenRepositoryUrls) {
+      RemoteRepository repository =
+          new RemoteRepository.Builder(null, "default", mavenRepositoryUrl).build();
+      if (!ALLOWED_REPOSITORY_URL_SCHEMES.contains(repository.getProtocol())) {
+        throw new IllegalArgumentException(
+            "Protocol: " + repository.getProtocol() + " is not in "
+                + ALLOWED_REPOSITORY_URL_SCHEMES);
+      }
+      try {
```

Comment:
Should check this at the top of the for loop before other work is done

Reply:
If you wanted to be fancy, you could check all the URLs and throw an exception that lists all the invalid ones in one message. Not necessary for now, but maybe nicer for the user.

Reply:
That's a nice idea. I think `ClassPathBuilder` can benefit from the idea more than this repository URLs. As of now  `ClassPathBuilder` fails immediately for one unavailable Maven artifact without telling how many failures need to be fixed.

Created #355.